### ISSUE TITLE
Syntax for fully qualified names ambiguous with field access

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/ergo#310]
It is sometimes useful to be able to use fully qualified names in the code. For instance:

```
new org.accordproject.helloworld.Request{ input : "hi" }
```
rather than:
```
import org.accordproject.helloworld.*
new Request{ input : "hi" }
```

when referring to global variables or functions, the fully qualified name syntax is ambiguous with the use of `.` for accessing the field of an object/record.

During parsing, a call to a fully qualified function as in:
```
foo.bar.baz(3)
```
might be easily confused with accessing the `bar` field of a `foo` variable:
```
foo.bar
``

**Comment by jeromesimeon** [comment]
Proposal to resolve this is to keep `foo.bar` for field access which is a very common operation, but have a special syntax for fully qualified names, which should be less commonly used (since one can `import`).

```
baz -- local name
@foo.bar.baz -- fully qualified name
foo.bar.baz -- accessing the fields of an object/record
```

A long discussion with @kach occurred to talk about other possible symbols that could be used instead of `@`.

```
@foo.bar.baz -- @ may be valuable for other things
#foo.bar.baz  -- 
$foo.bar.baz  -- $ may look too much like a variables
% ^ * / {}          -- already used
~foo.bar.baz  -- ??
`foo.bar.baz  -- ??
```

**Comment by kach** [comment]
I think `$` would be best here, but `@` is fine too (just means we can't use `@` elsewhere).

**Comment by jeromesimeon** [comment]
```
foo.bar.baz -- constant in the foo.bar namespace
foo.bar.baz -- accessing the baz field in the bar field foo
```

**Comment by jeromesimeon** [comment]
```
`foo.bar.baz`
'foo.bar.baz'
```

**Comment by jeromesimeon** [comment]
We decided to use `~foo.bar.baz` for fully qualified names in the language.

**Comment by jeromesimeon** [comment]
Implemented in #323
